src/app.rs

```diff
@@ -0,0 +1,138 @@
+use crate::cli::{Cli, Command};
+use crate::cloud_client::CloudClient;
+use crate::errors::PARSE_COMMAND_ERROR;
+use crate::tui::WorkMode;
+use crate::utilities::files::get_path_entries;
+use std::path::PathBuf;
+use tracing::debug;
+
+pub struct WorkspaceData {
+    pub local_path: PathBuf,
+    pub local_entries: Vec<String>,
+    pub cloud_path: PathBuf,
+    pub cloud_entries: Vec<String>,
+}
+
+impl Default for WorkspaceData {
+    fn default() -> Self {
+        Self {
+            local_path: std::env::current_dir().unwrap_or_default(),
+            local_entries: vec![],
+            cloud_path: PathBuf::from("//"),
+            cloud_entries: vec![],
+        }
+    }
+}
+
+pub struct App<C: CloudClient> {
+    pub input_command: String,
+    pub cursor_position: usize,
+    pub work_mode: WorkMode,
+    pub logs: Vec<String>,
```

Comment:
why do you need to save logs in a vector of strings?

Reply:
`logs` vector is used at tui rendering function `ui`

Reply:
@BigglesworthCat please, answer this one (I'm not telling you to fix it. just answer)

Reply:
it's better to write such logs in a file as the bash shell does (`~/.bash_history`). but I'm ok with your solution for now